test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs

```diff
@@ -3941,5 +3941,27 @@ public void PackCommand_PackProject_PacksFrameworkReferences_FrameworkReferences
                 }
             }
         }
+
+        [PlatformFact(Platform.Windows)]
+        public void PackCommand_WithGeneratePackageOnBuildSet_CanPublish()
+        {
+            using (var testDirectory = TestDirectory.Create())
+            {
+                var projectName = "ClassLibrary1";
+                var workingDirectory = Path.Combine(testDirectory, projectName);
+                // Act
+                msbuildFixture.CreateDotnetNewProject(testDirectory.Path, projectName, " console");
+
+                var projectFile = Path.Combine(workingDirectory, $"{projectName}.csproj");
+                using (var stream = new FileStream(projectFile, FileMode.Open, FileAccess.ReadWrite))
+                {
+                    var xml = XDocument.Load(stream);
+                    ProjectFileUtils.AddProperty(xml, "GeneratePackageOnBuild", "true");
+                    ProjectFileUtils.WriteXmlToFile(xml, stream);
+                }
+                // Run and assert.
+                msbuildFixture.RunDotnet(workingDirectory, $"publish {projectFile}");
```

Comment:
I don't know the `msbuildfixture` utility. What is it asserting? I guess that the command ran successfully.  Can it return the process exit code, so we can write the assert in the test, to make it more obvious to people reading the test?

Reply:
It asserts the exit code. 

By default it is asserting that it completed successfully. 

It also returns a command runner result. 

I can add another assert to make it clearer.